lldb/tools/lldb-dap/DAPLog.cpp

```diff
@@ -0,0 +1,31 @@
+//===-- DAPLog.cpp --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "DAPLog.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/raw_ostream.h"
+#include <chrono>
+#include <fstream>
+#include <mutex>
+#include <system_error>
+
+using namespace llvm;
+
+namespace lldb_dap {
+
+Log::Log(StringRef filename, std::error_code &EC) : m_stream(filename, EC) {}
+
+void Log::WriteMessage(StringRef message) {
+  std::scoped_lock<std::mutex> lock(m_mutex);
```

Comment:
Nit, can this be a `lock_guard`? I know a `scoped_lock` is more versatile but most of LLDB uses the former.

Reply:
Done.